demo/src/checkout-pricing.js

```diff
@@ -15,24 +15,41 @@ function CheckoutPricingForm () {
   const [recurlyError, setRecurlyError] = useState(null);
   const [plan, setPlan] = useState('');
   const [quantity, setQuantity] = useState('');
+  const [coupon, setCoupon] = useState('');
+  const [giftCard, setGiftCard] = useState('');
+  const [currency, setCurrency] = useState('');
+  const [billingCountry, setBillingCountry] = useState('');
+  const [billingPostalCode, setBillingPostalCode] = useState('');
+  const [billingVatNumber, setBillingVatNumber] = useState('');
+  const [shippingCountry, setShippingCountry] = useState('');
+  const [shippingPostalCode, setShippingPostalCode] = useState('');
+  const [shippingVatNumber, setShippingVatNumber] = useState('');
```

Comment:
Would it be better to track state in a larger blob and have the handlers call into it with a spread of the current state? Unsure if that's a no-no, but this is a lot of individual state methods to keep track of.

```jsx
const [billingAddress, setbillingAddress] = useState({
  country: '',
  postalCode: '',
  vatNumber: ''
});

// ...

<input
  type="text"
  value={billingAddress.country}
  onChange={e => setBillingAddress({ ...billingAddress, country: e.target.value })}
  placeholder="Country"
/>
```

Reply:
React's recommendation is to only bundle pieces of state together that update together:

> we recommend to split state into multiple state variables based on which values tend to change together.

https://reactjs.org/docs/hooks-faq.html#should-i-use-one-or-many-state-variables

Reply:
👍 good notes in that doc. I think it can improve readability to merge some state together, while tending to use single state updaters for most situations. I'm mostly concerned with making the demo as readable as possible (without impacting perf unnecessarily).

Agreed that the demo could use some improvement. I'd think that there may be a UI lib we could pull in to at least do away with the CSS.